src/invidious/channels/about.cr

```diff
@@ -43,36 +44,97 @@ def get_about_info(ucid, locale) : AboutChannel
     auto_generated = true
   end
 
-  if auto_generated
-    author = initdata["header"]["interactiveTabbedHeaderRenderer"]["title"]["simpleText"].as_s
-    author_url = initdata["microformat"]["microformatDataRenderer"]["urlCanonical"].as_s
-    author_thumbnail = initdata["header"]["interactiveTabbedHeaderRenderer"]["boxArt"]["thumbnails"][0]["url"].as_s
-
-    # Raises a KeyError on failure.
-    banners = initdata["header"]["interactiveTabbedHeaderRenderer"]?.try &.["banner"]?.try &.["thumbnails"]?
-    banner = banners.try &.[-1]?.try &.["url"].as_s?
+  tab_names = [] of String
+  total_views = 0_i64
+  joined = Time.unix(0)
 
-    description_node = initdata["header"]["interactiveTabbedHeaderRenderer"]["description"]
+  if ageGate = initdata.dig?("contents", "twoColumnBrowseResultsRenderer", "tabs", 0, "tabRenderer", "content", "sectionListRenderer", "contents", 0, "channelAgeGateRenderer")
+    description_node = nil
+    author = ageGate["channelTitle"].as_s
+    ucid = initdata["responseContext"]["serviceTrackingParams"][0]["params"][0]["value"].as_s
+    author_url = "https://www.youtube.com/channel/" + ucid
+    author_thumbnail = ageGate["avatar"]["thumbnails"][0]["url"].as_s
+    banner = nil
+    is_family_friendly = false
+    is_age_gated = true
+    tab_names = ["videos", "shorts", "streams"]
+    auto_generated = false
   else
-    author = initdata["metadata"]["channelMetadataRenderer"]["title"].as_s
-    author_url = initdata["metadata"]["channelMetadataRenderer"]["channelUrl"].as_s
-    author_thumbnail = initdata["metadata"]["channelMetadataRenderer"]["avatar"]["thumbnails"][0]["url"].as_s
-    author_verified = has_verified_badge?(initdata.dig?("header", "c4TabbedHeaderRenderer", "badges"))
+    if auto_generated
```

Comment:
can you use `elsif` here? That'd greatly reduce the diff's size!

Reply:
It should be possible but lines _97_ to _136_ are outside of the nested `if-else` so it'd require another if statement

Reply:
Ohhh, right, forgot about that, thanks!